### PR TITLE
PUBLIC-2591: Python 3.12 update

### PR DESCRIPTION
**Pull request by ialarmedalien**
Updating the docker file to use python 3.12 on bookworm.
Cleaning up various bits of cruft.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 
 WORKDIR /kb/module
-# Update
 RUN apt-get update \
     && apt-get upgrade -y && \
     update-ca-certificates && \
     apt-get -y install xvfb curl && \
+    # these are all needed by PyQt5
+    apt-get -y install libdbus-1-3 libxcb-keysyms1 libxcb-image0 libxkbcommon-x11-0 libxkbcommon0 libxcb-icccm4 libxcb-image0 libxcb-render-util0 libxcb-shape0 libxcb-xinerama0 && \
+    apt-get clean && \
```
Comment (ialarmedalien) [inline Dockerfile:11]: one of the ete3 dependencies requires a whole load of system packages that weren't installed (even as part of the standard/non-slim bookworm version).

File: scripts/prepare_deploy_cfg.py [hunk 2]
```diff
@@ -1,21 +1,44 @@
+"""Generate the deployment config file."""
```
Comment (ialarmedalien) [inline scripts/prepare_deploy_cfg.py:1]: I copied this from the Static Narrative repo as the existing version didn't allow you to use the `KBASE_ENDPOINT` env var.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ete3==3.1.3
+jinja2==3.1.4
+jsonrpcbase==0.2.0
+numpy==2.0.1
+PyQt5==5.15.11
+pytest==8.3.2
+pytest-cov==5.0.0
+pytest-recording==0.13.2
+requests==2.32.3
+ruff==0.5.7
```
Comment (ialarmedalien) [inline requirements.txt:11]: there's only one dev requirement so I just included it in the main requirements file 🤷

**Review by briehl — APPROVED** [review]
LGTM

**Pull request merged**